workspace/src/app/views/shared/modals/CreateArtefactModal/ArtefactForms/ParameterAutoCompletion.tsx

```diff
@@ -168,13 +171,44 @@ export const ParameterAutoCompletion = ({
 
     const itemValue = (value: StringOrReifiedValue) => (typeof value === "string" ? value : value.value);
 
+    const handleSearch = React.useCallback(
+        (input: string) => {
+            setSearchQuery(input);
+            setFetchMore(true)
+            setLimit(100)
+            return handleAutoCompleteInput(input, autoCompletion);
+        },
+        [limit]
+    );
+
+    const loadMoreResults = async () => {
+        const newLimit = limit + 100;
+        const results = (await handleAutoCompleteInput(searchQuery, autoCompletion, newLimit)) ?? [];
+        setLimit(newLimit);
+
+        //make one more requests if the response is less than the limit
+        if (results.length < newLimit) {
+            //hide loadMore
+            setFetchMore(false);
+        } else {
+            //fetch one more ahead to see if there is still more.
+            const nextLimit = newLimit + 1;
+            const nextResults = (await handleAutoCompleteInput(searchQuery, autoCompletion, nextLimit)) ?? [];
```

Comment:
You could combine the +1 into the previous request (leaving out the additional item) to only make 1 request instead of 2.